src/MiniExec.sol

```diff
@@ -35,71 +35,37 @@ contract MiniExecImplementation is IZkEVMBridgeMessageReceiver {
             }
         }
     }
-
-    // @OF: see comment in constructor - that would avoid making a call to the factory
-    // @OF: if this logic is kept, I think it's better to inline this?
-    function _metadata() internal view returns (address, uint32) {
-        return creator.metadata(address(this));
-    }
 }
 
-contract MiniExecProxy is Proxy {
+contract MiniExecProxy is Proxy, MiniExecStorage {
     error OnlySelf();
 
     MiniExecFactory immutable creator;
 
-    // @OF: spec says "Constructor takes in tuple of address and chain ID"
-    constructor() {
-        creator = MiniExecFactory(msg.sender);
-    }
-
-    function updateImplementation(address _newImplementation) external onlySelf {
-        creator.setImplementation(_newImplementation);
+    constructor(address owner, uint32 networkId, address impl) {
```

Comment:
note: while this works, the common pattern for setting the values in the implementation is to pass the values through an initializer

Reply:
yeah but comment/spec was specifically for a constructor?